### ISSUE TITLE
Indentation rule stopped working after upgrading to version 1.10.0

### ISSUE THREAD

**Issue by keymusicman** [detekt/detekt#2970]
## Expected Behavior
In our project we have Indentation rule turned on with the following rule settings:

```
  Indentation:
    active: true
    autoCorrect: true
    indentSize: 2
    continuationIndentSize: 4
```
It is done in accord with our project settings:
![image](https://user-images.githubusercontent.com/30913903/90242883-8047a280-de36-11ea-90c4-a287f572add7.png)

It worked prior to version 1.10.0

## Observed Behavior
We tried upgrading detekt to version 1.10.0 and got 13k errors related to Indentation rule. It says we have wrong indent and the auto correction turned on tries to reformat the code.

The example code that fails Indentation rule after upgrading:
```kotlin
  override fun getAddresses() = async {
    addressRepository.getAll()
        .toVO { address -> address.id == id }
  }

```
Detekt says that the problem is on the line `.toVO { address -> address.id == id }`

I know that the new version of KtLint now uses a new Indentation rule that was in experimental state earlier, but I found no mentions that something should break and no guidelines on how to fix it.

## Your Environment

* Version of detekt used: 1.10.0
* Version of Gradle used (if applicable): 6.4
* Operating System and version: MacOSX 10.15.4 Catalina
* Link to your project (if it's a public repository): -

**Comment by cortinico** [comment]
> We tried upgrading detekt to version 1.10.0 and got 13k errors related to Indentation rule. It says we have wrong indent and the auto correction turned on tries to reformat the code.

> I know that the new version of KtLint now uses a new Indentation rule that was in experimental state earlier, but I found no mentions that something should break and no guidelines on how to fix it.

As you said, you're not alone: https://github.com/pinterest/ktlint/issues/764

The current workaround is either to:
- Disable the `Indentation` rule at all.
- Disable the `Indentation` rule at all + upgrade detekt and run an older version of `ktlint` that uses the old `indentation` rule. 
- Don't upgrade and wait for an updated version of KtLint + an updated version of Detekt.

**Comment by schalkms** [comment]
Shall we close this issue then in order to not have a duplicate?

**Comment by keymusicman** [comment]
Yes, it can be closed. As the problem is actually related to KtLint, i will follow its ticket (thanks @cortinico !)

**Comment by keymusicman** [comment]
The issue seems should be fixed in KtLint `0.41`. I've just upgraded to Detekt `1.19.0` that uses KtLint 0.42.1 and got the same issue - still does not work

**Comment by cortinico** [comment]
> still does not work

Can you provide a reproducer?

**Comment by calvarez-ov** [comment]
Here's a bug reproduction project: https://github.com/libon/DetektIndentationIssue

**Comment by cortinico** [comment]
Thanks for the reproducer @calvarez-ov 
Turns out that KtLint removed the `continuation_indent_size` in a older version. See:
- https://github.com/pinterest/ktlint/issues/171
- https://github.com/pinterest/ktlint/issues/308

Therefore we need to deprecate the following config as it's not taken into account by KtLint at all:

https://github.com/detekt/detekt/blob/e0fa7a3e81312d6edf052b13efc82998c59972a6/detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/wrappers/Indentation.kt#L28-L29
